File: php/src/Models/BigQueryConfig.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Models;

/** Configuration for a Google Cloud BigQuery sink. */
class BigQueryConfig implements \JsonSerializable
{
    private array $setFields = [];

    /**
     * @param string $credentials google Cloud Credentials JSON Object as a string
     */
    private function __construct(
        public readonly string $projectId,
        public readonly string $datasetId,
        public readonly string $tableId,
        public readonly string $credentials,
        array $setFields = [],
    ) {
        $this->setFields = $setFields;
    }

    /**
     * Create an instance of BigQueryConfig with required fields.
     */
    public static function create(
        string $projectId,
        string $datasetId,
        string $tableId,
        string $credentials,
    ): self {
        return new self(
            projectId: $projectId,
            datasetId: $datasetId,
            tableId: $tableId,
            credentials: $credentials,
            setFields: ['projectId' => true, 'datasetId' => true, 'tableId' => true, 'credentials' => true]
        );
    }

    public function jsonSerialize(): mixed
    {
        $data = [
            'projectId' => $this->projectId,
            'datasetId' => $this->datasetId,
            'tableId' => $this->tableId,
            'credentials' => $this->credentials];

        return \Svix\Utils::newStdClassIfArrayIsEmpty($data);
    }

    /**
     * Create an instance from a mixed obj.
     */
    public static function fromMixed(mixed $data): self
    {
        return new self(
            projectId: \Svix\Utils::deserializeString($data, 'projectId', true, 'BigQueryConfig'),
            datasetId: \Svix\Utils::deserializeString($data, 'datasetId', true, 'BigQueryConfig'),
            tableId: \Svix\Utils::deserializeString($data, 'tableId', true, 'BigQueryConfig'),
            credentials: \Svix\Utils::deserializeString($data, 'credentials', true, 'BigQueryConfig')
        );
    }

    /**
     * Create an instance from a json string.
     */
    public static function fromJson(string $json): self
    {
        $data = json_decode(json: $json, associative: true, depth: 512, flags: JSON_THROW_ON_ERROR);

        return self::fromMixed($data);
    }
}

File: java/lib/src/main/java/com/svix/models/BigQueryConfig.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.svix.Utils;

import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(getterVisibility = Visibility.NONE, setterVisibility = Visibility.NONE)
public class BigQueryConfig {
    @JsonProperty private String projectId;
    @JsonProperty private String datasetId;
    @JsonProperty private String tableId;
    @JsonProperty private String credentials;

    public BigQueryConfig() {}

    public BigQueryConfig projectId(String projectId) {
        this.projectId = projectId;
        return this;
    }

    /**
     * Get projectId
     *
     * @return projectId
     */
    @javax.annotation.Nonnull
    public String getProjectId() {
        return projectId;
    }

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public BigQueryConfig datasetId(String datasetId) {
        this.datasetId = datasetId;
        return this;
    }

    /**
     * Get datasetId
     *
     * @return datasetId
     */
    @javax.annotation.Nonnull
    public String getDatasetId() {
        return datasetId;
    }

    public void setDatasetId(String datasetId) {
        this.datasetId = datasetId;
    }

    public BigQueryConfig tableId(String tableId) {
        this.tableId = tableId;
        return this;
    }

    /**
     * Get tableId
     *
     * @return tableId
     */
    @javax.annotation.Nonnull
    public String getTableId() {
        return tableId;
    }

    public void setTableId(String tableId) {
        this.tableId = tableId;
    }

    public BigQueryConfig credentials(String credentials) {
        this.credentials = credentials;
        return this;
    }

    /**
     * Google Cloud Credentials JSON Object as a string.
     *
     * @return credentials
     */
    @javax.annotation.Nonnull
    public String getCredentials() {
        return credentials;
    }

    public void setCredentials(String credentials) {
        this.credentials = credentials;
    }

    /**
     * Create an instance of BigQueryConfig given an JSON string
     *
     * @param jsonString JSON string
     * @return An instance of BigQueryConfig
     * @throws JsonProcessingException if the JSON string is invalid with respect to BigQueryConfig
     */
    public static BigQueryConfig fromJson(String jsonString) throws JsonProcessingException {
        return Utils.getObjectMapper().readValue(jsonString, BigQueryConfig.class);
    }

    /**
     * Convert an instance of BigQueryConfig to an JSON string
     *
     * @return JSON string
     */
    public String toJson() throws JsonProcessingException {
        return Utils.getObjectMapper().writeValueAsString(this);
    }
}
